### PR TITLE
feature/add-bbox-kwarg

### PR DESCRIPTION
**Pull request by ritvikvasan**
- [x] hard coded bounding box kwarg (ZYX) to pass in. For example [64, 168, 104]

### CODE REVIEW

**Comment by evamaxfield** [comment]
Can you rebase this off of `master` branch? You branched off of your branch and I think this one is easier to merge than the diagnostics sheets probably.

File: actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+        distributed_executor_address: Optional[str]
+            An optional executor address to pass to some computation engine.
+        clean: bool
+            Should the local staging directory be cleaned prior to this run.
+            Default: False (Do not clean)
+        debug: bool
+            A debug flag for the developer to use to manipulate how much data runs,
+            how it is processed, etc.
+            Default: False (Do not debug)
+
```
Comment (donovanr) [inline actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py]: @JacksonMaxfield do you recommend removing this?

File: actk/steps/single_cell_images/single_cell_images.py [hunk 2]
```diff
@@ -333,6 +337,12 @@ def run(
             # Block until all complete
             bbox_results = handler.gather(bounding_box_futures)
 
+            if bbox:
+                num_of_channels = bbox_results[0][0]
+                bbox = [num_of_channels] + list(bbox)
+                bbox_results = [bbox for i in range(len(bbox_results))]
+                log.info(f"Using hard coded bounding box with ZYX dimensions: {bbox}.")
+
```
Comment (donovanr) [inline actk/steps/single_cell_images/single_cell_images.py:348]: Is this the only place the `bbox` parameter get used?  I'm unclear on how the cropping happens -- to me 
```bbox_results = [bbox for i in range(len(bbox_results))]```
looks like a `len(bbox_results)` list of just the same tuple, e.g. `[(64, 168, 104), (64, 168, 104), ... ]`

**Review by donovanr — CHANGES_REQUESTED** [review]
LIke @JacksonMaxfield suggested, lets disentangle the bbox changes from the diagnostic sheet stuff so they can merge to mater easier and what exactly is getting changed is more clear.

File: actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py (hunk 1 above)
Comment (evamaxfield) [reply actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py]: Check other PR just commented on all of this.

File: actk/steps/single_cell_images/single_cell_images.py (hunk 2 above)
Comment (ritvikvasan) [reply actk/steps/single_cell_images/single_cell_images.py:348]: I think the cropping is happening via aicsimageprocessing.cell_rigid_registration - 

https://github.com/AllenCellModeling/actk/blob/1d4ae5fc01b62b9bb35789aae5efa72e46d4b6a5/actk/steps/single_cell_images/single_cell_images.py#L142-L147, 

and the bounding box is computed here (averaged across all registered image sizes) -
https://github.com/AllenCellModeling/actk/blob/1d4ae5fc01b62b9bb35789aae5efa72e46d4b6a5/actk/steps/single_cell_images/single_cell_images.py#L339-L342

before being passed into the distributed handler via _generate_single_cell_images. So I thought the best place to add the bbox argument would be here. I am broadcasting bbox across length len(bbox_results) to be consistent with the bbox_result shape collected without the hard coded box parameters.

Let me know if this doesn't make sense and I should change this!

**Comment by donovanr** [comment]
I fixed the merge conflict from the diagnostic sheet merge, basically just manually merged all the incoming changes over what's in the current branch because they were in an orthogonal step.  hopefully didn't screw anything up too badly, but i was hoping to get all these changes in master so we can get this off our plate.

**Pull request merged**